test/test_integration.py

```diff
@@ -236,19 +534,29 @@ def test_loading_slider(self):
 
         @app.callback(
             Output('horizontal-slider', 'value'),
-            [Input('test-div', 'children')]
+            [Input('test-btn', 'n_clicks')]
         )
-        def delayed_value(children):
-            lock.acquire()
-            lock.release()
-            return 5
+        def user_delayed_value(n_clicks):
+            with lock:
+                return 5
 
-        self.startServer(app)
+        with lock:
+            self.startServer(app)
+
+            self.wait_for_element_by_css_selector(
+                '#horizontal-slider[data-dash-is-loading="true"]'
+            )
 
         self.wait_for_element_by_css_selector(
-            '#horizontal-slider[data-dash-is-loading="true"]'
+            '#horizontal-slider:not([data-dash-is-loading="true"])'
         )
-        lock.release()
+
+        with lock:
+            self.driver.find_element_by_id('test-btn').click()
+
+            self.wait_for_element_by_css_selector(
+                '#horizontal-slider[data-dash-is-loading="true"]'
+            )
```

Comment:
Adding to the test -- we don't only check that the loading state is ok on initialization, also check that it's ok on user initiated actions